### PR TITLE
Add DrizzleORM to the backend

### PR DESCRIPTION
**Pull request by fracek**
This PR adds DrizzleORM to the backend, so that we can use it to generate the schema migrations going forward.

It also has a nice API for building the backend API.

We need a follow-up PR to split the docker-compose file in `infra/indexers` in two: one with the
database (also used by the backend) and one with just the indexers.

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
 
 ```bash
 bun run index.ts
 ```
 
-This project was created using `bun init` in bun v1.0.18. [Bun](https://bun.sh) is a fast all-in-one JavaScript runtime.
+**Updating the database schema**
+
+1. Update the tables in `src/db/schema.ts`.
+2. Run `npx drizzle-kit generate:pg` to generate the SQL migrations.
```
Comment (LucasLvy) [inline backend/README.md:20]:
```suggestion
2. Run `bunx drizzle-kit generate:pg` to generate the SQL migrations. 
```

File: backend/src/db/migrations/0000_dry_silver_centurion.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE IF NOT EXISTS "transfer_usdc" (
```
Comment (LucasLvy) [inline backend/src/db/migrations/0000_dry_silver_centurion.sql:1]: the names are weird is that expected to be pushed ?

File: backend/src/db/migrations/meta/0000_snapshot.json [hunk 3]
```diff
@@ -0,0 +1,91 @@
+{
```
Comment (LucasLvy) [inline backend/src/db/migrations/meta/0000_snapshot.json:1]: what is that ?

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: backend/README.md (hunk 1 above)
Comment (fracek) [reply backend/README.md:20]: Not a typo, that command doesn't work with bun yet :(

File: backend/src/db/migrations/0000_dry_silver_centurion.sql (hunk 2 above)
Comment (fracek) [reply backend/src/db/migrations/0000_dry_silver_centurion.sql:1]: Yep, they're the migrations to be applied to the database.

File: backend/src/db/migrations/meta/0000_snapshot.json (hunk 3 above)
Comment (fracek) [reply backend/src/db/migrations/meta/0000_snapshot.json:1]: It's needed by drizzle to generate migrations.

File: backend/README.md (hunk 1 above)
Comment (LucasLvy) [reply backend/README.md:20]: works on my machine:
```
 bunx drizzle-kit generate:pg
drizzle-kit: v0.20.14
drizzle-orm: v0.30.7

No config path provided, using default 'drizzle.config.ts'
Reading config file '/Users/lucas/vault/backend/drizzle.config.ts'
2 tables
balance_usdc 6 columns 0 indexes 0 fks
transfer_usdc 11 columns 0 indexes 0 fks

No schema changes, nothing to migrate 😴
```

Comment (fracek) [reply backend/README.md:20]: Good to know, I will change it to `bunx` then!

**Comment by fracek** [comment]
I updated the PR.

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**